server/base/src/main/java/org/apache/accumulo/server/conf/ServerConfigurationFactory.java

```diff
@@ -99,10 +144,139 @@ public NamespaceConfiguration getNamespaceConfigurationForTable(TableId tableId)
+        "configuration snapshot refresh completed. Total runtime {} ms for local namespaces: {}, tables: {}",
+        MILLISECONDS.convert(System.nanoTime() - refreshStart, NANOSECONDS), nsRefreshCount,
+        tblRefreshCount);
+  }
+
+  private boolean refresh(ZooBasedConfiguration config, ZooReader zooReader) {
+    final PropStoreKey<?> key = config.getPropStoreKey();
+    try {
+      Stat stat = zooReader.getStatus(key.getPath());
+      log.trace("configuration snapshot refresh: stat returned: {} for {}", stat, key);
+      if (stat == null) {
+        return false;
+      }
+      if (config.getDataVersion() != stat.getVersion()) {
+        log.debug(
+            "configuration snapshot refresh - difference found. forcing configuration update for {}}",
+            key);
+        config.zkChangeEvent(key);
+      }
+      // add small jitter between calls.
+      int randDelay = ThreadLocalRandom.current().nextInt(0, 23);
+      Thread.sleep(randDelay);
+    } catch (KeeperException.NoNodeException ex) {
+      config.zkChangeEvent(key);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("Interrupted reading from ZooKeeper during snapshot refresh.",
+          ex);
+    } catch (KeeperException | IllegalStateException ex) {
+      log.debug("configuration snapshot refresh: Exception during refresh - key" + key, ex);
+    }
+    return true;
+  }
+
+  private static class DeleteWatcher implements PropChangeListener {
+
+    final Map<TableId,TableConfiguration> tableConfigs;
+    final Map<NamespaceId,NamespaceConfiguration> namespaceConfigs;
+    final Map<TableId,NamespaceConfiguration> tableParentConfigs;
+
+    DeleteWatcher(final Map<TableId,TableConfiguration> tableConfigs,
+        final Map<NamespaceId,NamespaceConfiguration> namespaceConfigs,
+        final Map<TableId,NamespaceConfiguration> tableParentConfigs) {
+      this.tableConfigs = tableConfigs;
+      this.namespaceConfigs = namespaceConfigs;
+      this.tableParentConfigs = tableParentConfigs;
+    }
+
+    @Override
+    public void zkChangeEvent(PropStoreKey<?> propStoreKey) {
+      // no-op. changes handled by prop store impl
+    }
+
+    @Override
+    public void cacheChangeEvent(PropStoreKey<?> propStoreKey) {
+      // no-op. changes handled by prop store impl
+    }
+
+    @Override
+    public void deleteEvent(PropStoreKey<?> propStoreKey) {
+      if (propStoreKey instanceof NamespacePropKey) {
+        log.trace("configuration snapshot refresh: Handle namespace delete for {}", propStoreKey);
+        namespaceConfigs.remove(((NamespacePropKey) propStoreKey).getId());
+        return;
+      }
+      if (propStoreKey instanceof TablePropKey) {
+        log.trace("configuration snapshot refresh: Handle table delete for {}", propStoreKey);
+        tableConfigs.remove(((TablePropKey) propStoreKey).getId());
+        tableConfigs.remove(((TablePropKey) propStoreKey).getId());
```

Comment:
This is removed twice. I think the intent was:

```suggestion
        tableConfigs.remove(((TablePropKey) propStoreKey).getId());
        tableParentConfigs.remove(((TablePropKey) propStoreKey).getId());
```